shapeshift/web
# chore: useswapper cleanup

## Description

Cleanup useSwapper useages in web

- Swappers are now assumed to always use sellAmount -- simplifies logic in many places at the expense of slightly less accuracy (approved by beard). Swapper PR to follow removing logic there as well
- Differentiate between "trade" and "quote" and use the correct values in correct locations.
- getQuote flow has been entirely re-written to be far simpler and now correctly debounces
- Lots of removed unnecesary code & renames to make more sense

This PR might be easier to understand by looking at swapper as a whole instead of trying to figure out the diff. a lot has changed.

## Notice




## Pull Request Type


## Issue (if applicable)

https://github.com/shapeshift/web/issues/1689

## Risk

Trading could be borked if I messed something u[p

## Testing

Make sure that all aspects of swapping behave the same as before. Confirm that everything looks correct and loads prices for for:

- Selecting buy asset
- Selecting sell assert
- Toggle sell asset with buy asset
- Go to confirm screen with a token that will need approval and see that it correctly shows approval tx
- Go to confirm screen and see that all values make sense
- Do a trade and see it work
